### DISCUSSION TITLE
Dynamic Object Members Validation

### DISCUSSION THREAD

**Discussion by WhitestRabbit** [adonisjs/core#2853] [Help]
I have a validation case where I want to be given a field called 'options' (from a JSON or form) which can look something like this:

"options": { "2": "", "5": { "value": "foo"}, "3": {"value": "bar"}  ...}

Essentially, I want to receive an object with no fixed amount of entries, where the keys of the entries can be positive numbers and the values of the entries can either be an empty string, or an object that contains a single entry, with a "value" key and a string for the actual entry value.

I think I've managed to succeed in the second part of the validation. Is there any way to implement the "no fixed amount of entries, where the keys are positive numbers" part?

**Comment by arthurcvm** [comment]
Hi @WhitestRabbit, use [Custom rule](https://github.com/adonisjs/legacy.adonisjs.com/blob/develop/docs/docs/guides/validator/08-adding-custom-rules.md) to do this.

**Reply by WhitestRabbit** [reply]
it is more of a custom thing so yep

**Reply by arthurcvm** [reply]
@WhitestRabbit  if works pls mark this solution.
